tfhe/src/high_level_api/compact_list.rs

```diff
@@ -78,26 +87,54 @@ impl CompactCiphertextList {
                     IntegerCompactCiphertextListCastingMode::NoCasting
                 };
 
-                self.0
+                self.inner
                     .expand(unpacking_mode, casting_mode)
-                    .map(|inner| CompactCiphertextListExpander { inner })
+                    .map(|inner| CompactCiphertextListExpander {
+                        inner,
+                        tag: self.tag.clone(),
+                    })
             }
             #[cfg(feature = "gpu")]
             Some(_) => Err(crate::Error::new("Expected a CPU server key".to_string())),
         })
     }
 }
+
+impl Tagged for CompactCiphertextList {
+    fn tag(&self) -> &Tag {
+        &self.tag
+    }
+
+    fn tag_mut(&mut self) -> &mut Tag {
+        &mut self.tag
+    }
+}
+
 impl ParameterSetConformant for CompactCiphertextList {
     type ParameterSet = CompactCiphertextListConformanceParams;
 
     fn is_conformant(&self, parameter_set: &Self::ParameterSet) -> bool {
-        self.0.is_conformant(parameter_set)
+        self.inner.is_conformant(parameter_set)
     }
 }
 
 #[cfg(feature = "zk-pok")]
 #[derive(Clone, Serialize, Deserialize)]
-pub struct ProvenCompactCiphertextList(crate::integer::ciphertext::ProvenCompactCiphertextList);
+pub struct ProvenCompactCiphertextList {
+    inner: crate::integer::ciphertext::ProvenCompactCiphertextList,
+    tag: Tag,
+}
+
+#[cfg(feature = "zk-pok")]
+impl Tagged for ProvenCompactCiphertextList {
```

Comment:
should we create a zk mod here and have a feature gate on it + a pub re export ?

allows to limit the number of #cfg and I find it less error prone